src/functional/ffront/foast_passes/type_deduction.py

```diff
@@ -224,12 +224,12 @@ def visit_Assign(self, node: foast.Assign, **kwargs) -> foast.Assign:
         new_target = self.visit(node.target, refine_type=new_value.type, **kwargs)
         return foast.Assign(target=new_target, value=new_value, location=node.location)
 
-    def visit_FieldSymbol(
+    def visit_Symbol(
         self,
-        node: foast.FieldSymbol,
+        node: foast.Symbol,
         refine_type: Optional[common_types.FieldType] = None,
         **kwargs,
-    ) -> foast.FieldSymbol:
+    ) -> foast.Symbol:
```

Comment:
I guess for static type checkers it would be better to define two overloads here, one when `refine_type` is `None` which returns a parameterized `foast.Symbol[T]` using the same `TypeVar` than the argument, and another with non-optional `refine_type` `TypeVar("RefinedT", common_types.FieldType, common_types.TupleType)` value which returns a parameterized `foast.Symbol[RefinedT]`.

I'm not sure mypy will be able to deal with all the Nodes/pydantic magic anyway, so maybe we can skip it for now, but at least I want to document the fact.

Reply:
I agree. Skipping for now.